cmd/sanitize-prow-jobs/main.go

```diff
@@ -140,6 +140,9 @@ func main() {
 	if err != nil {
 		logrus.WithError(err).Fatalf("Failed to load config from %q", opt.configPath)
 	}
+	if err := config.AreDeterministic(opt.prowJobConfigDir); err != nil {
+		logrus.WithError(err).Fatal("Failed to check if the config can determinize jobs")
+	}
```

Comment:
So this ends up doing a second full walkthrough? That sounds excessive.

Perhaps we can, while sanitizing a single job, walk through the whole config instead of selecting the first match? We would always iterate through all possibilities and if more than one matched, we would record a failure.

Reply:
Let me try.

Reply:
Updated.

Reply:
Why github does not show `Outdated`? I updated the PR.